subprojects/file-watching/src/main/java/org/gradle/internal/watch/vfs/impl/WatchingVirtualFileSystem.java

```diff
@@ -125,6 +125,9 @@ public SnapshotHierarchy call(BuildOperationContext context) {
                             );
                         }
                     }
+                    if (watchRegistry != null) {
+                        watchRegistry.setDebugLoggingEnabled(debugWatchLogging);
```

Comment:
Doesn't the Gradle daemon reset the log level at the end of the build?

Reply:
Checking the daemon log now.

Reply:
It resets the top-level logging, not `NativeLogging`'s level, so it is kept as is.